src/rpdk/core/init.py

```diff
@@ -139,14 +138,39 @@ def init(args):
 
     check_for_existing_project(project)
 
-    type_name = input_typename()
+    if args.type_name:
+        try:
+            type_name = validate_type_name(args.type_name)
+        except WizardValidationError as error:
+            print(Style.BRIGHT, Fore.RED, str(error), Style.RESET_ALL, sep="")
+            type_name = input_typename()
+    else:
+        type_name = input_typename()
+
     if args.language:
-        language = args.language
-        LOG.warning("Language plugin '%s' selected non-interactively", language)
+        language = args.language.lower()
+        if language not in get_plugin_choices():
+            print(
+                Style.BRIGHT,
+                Fore.RED,
+                "The plugin for {} is not installed.".format(language),
+                Style.RESET_ALL,
+                sep="",
+            )
+            language = input_language()
     else:
         language = input_language()
 
-    project.init(type_name, language)
+    project.init(
+        type_name,
+        language,
+        {
+            "use_docker": args.use_docker,
+            "namespace": args.namespace,
+            "codegen_template_path": args.codegen_model,
```

Comment:
At the moment, each plugin looks for the arguments it needs in `project.settings`, the fact that additional un-related information is also coming in does not affects the plugin. We sure can do some validation and only allow arguments related to the selected plugin but I don't know if is worth doing that validation or how it will impact the user experience, if the user passes a wrong flag s/he would need to try again (kind of defeating the purpose of being able to use just a one-liner). What do you think?